src/actions/sets.ts

```diff
@@ -255,8 +296,360 @@ export const renameSetPresetOnRemote = (preset: PresetRecord, newname: string):
 			dispatch(showNotification({
 				level: NotificationLevel.error,
 				title: `Error while trying to rename preset ${preset.name} to ${newname}`,
-				message: "Please check the consolor for further details."
+				message: "Please check the console for further details."
+			}));
+			console.log(err);
+		}
+	};
+
+export const initSetViews = (viewState?: OSCQueryRNBOSetViewListState): IInitGraphSetViews => {
+
+	const views: GraphSetViewRecord[] = [];
+	for (const [id, view] of Object.entries(viewState?.CONTENTS || {}) as Array<[string, OSCQueryRNBOSetView]>) {
+		views.push(
+			GraphSetViewRecord.fromDescription(id, view)
+		);
+	}
+
+	return {
+		type: GraphSetActionType.INIT_SET_VIEWS,
+		payload: {
+			views
+			// names.map(n => PresetRecord.fromDescription(n, n === "initial"))
+		}
+	};
+};
+
+export const setSetView = (view: GraphSetViewRecord): ISetGraphSetView => {
+	return {
+		type: GraphSetActionType.SET_SET_VIEW,
+		payload: {
+			view
+		}
+	};
+};
+
+export const createSetViewOnRemote = (name: string): AppThunk =>
+	(dispatch, getState) => {
+		try {
+			const state = getState();
+			const params = getPatcherInstanceParameters(state);
+			// TODO: ensure name is unique
+
+			const message = {
+				address: "/rnbo/inst/control/sets/views/create",
+				args: [
+					{ type: "s", value: name },
+					...params.valueSeq().map(p => ({ type: "s", value: p.setViewId })).toArray()
```

Comment:
this seems to add parameters sort of randomly, i think they should be sorted by `instance_index` then `index` just like you get when you do the `Include All` in the set view param editor